### PR TITLE
Relational: Make  `is_ge` handle extended real numbers properly

### LINKED ISSUE #10158: solve(x*Max(x, 15) - 10, x) doesn't work
`solve(x*Max(x, 15) - 10, x)` gives NotImplementedError. `solveset` can't solve it either. 

I think to solve equations like this, one should replace the Max with each argument and solve independently, then backtrack and remove any solutions where the Max argument is not actually the maximum. For instance, in this case, one would solve `x**2 - 10` and `x*15 - 10`. The first one gives `[sqrt(10), -sqrt(10)]`, but both of those solutions are invalid because they aren't the maximum in `Max(x, 15)`. The second gives `2/3`, which is the correct solution.

### PR DESCRIPTION
**Pull request by JosephMehdiyev**
#### References to other Issues or PRs
Fixes [#29413](https://github.com/sympy/sympy/issues/29413) and properly fixes https://github.com/sympy/sympy/issues/10158

#### Brief description of what is fixed or changed
Changed `is_ge` function from `relational.py` to handle the infinite and finite cases properly.

#### Other comments
The tests that are failing seems like were technically wrong anyways. We do NOT know if `x` is extended real or not, so saying `x < oo` implies x cannot be extended real, which is not true in these test cases.
(Assume `from sympy.abc import x, y, a etc.` unless stated otherwise)
Note that initializing variables as reals (or extended reals), makes them equivalent anyways. Seems like `x < oo` etc are not intended.
Here are some before the PR and after PR pictures that makes the tests fail:
```python
solve(abs(x) + abs(x - 1) - 1 > 0, x)
```
image
image

```python
u = Piecewise((0, x <= 0), (1, x >= a), (x/a, True))
e = (u - u.subs(x, y))**2/(x - y)**2
integrate(e,  x).expand()

```

image
image

```python
f = Function('f')
p = Piecewise((1, Eq(f(x) - 1, 0)), (2, x - 10 < 0), (0, True))
p.integrate(y)
```
image
image

If the maintainers are OK with changing the tests, I can manually check all 22 cases that are failing and change them accordingly IF the issue is related to ` x < oo` of course.
#### AI Generation Disclosure

NO AI USE

#### Release Notes

* core
  * Fixed a bug of relational `is_ge `to handle the infinite cases when the variables are unknown to be extended reals.

### CODE REVIEW

**Comment by JosephMehdiyev** [comment]
Here are the weird things I have found (and the tests that are properly failing)
Now, this actually works (See [Issue##10158](https://github.com/sympy/sympy/issues/10158))

```Python
solve(x*Max(x, 15) - 10, x) # gives 2/3
```

The functions `reduce_abs_inequality`, `reduce_rational_inequalities`  did not get affected by this change. It keeps `x < oo` stuff in its outputs

```Python
dom = S.Reals
solveset(Abs(x - 1) - Abs(y), x, dom) 
```
image
Becomes
image

However,

```Python
solve(Abs(x - 1) - Abs(y), x) #The output is consistent with the change
```

Also this PR completely breaks the 2 stat tests that are failing

**Comment by JosephMehdiyev** [comment]
There are still 3 tests failing. Two of them are statistics tests and I have no idea why this change breaks it completely.
The 3rd one is the above picture with `solveset(Abs(x - 1) - Abs(y), x, dom) `

File: sympy/core/relational.py [hunk 1]
```diff
@@ -1425,13 +1424,33 @@ def is_ge(lhs, rhs, assumptions=None):
         _lhs = AssumptionsWrapper(lhs, assumptions)
         _rhs = AssumptionsWrapper(rhs, assumptions)
         if _lhs.is_extended_real and _rhs.is_extended_real:
-            if (_lhs.is_infinite and _lhs.is_extended_positive) or (_rhs.is_infinite and _rhs.is_extended_negative):
+            if _lhs.is_infinite:
+                if _rhs.is_infinite:
+                    return _lhs.is_extended_positive or (not _rhs.is_extended_positive)
```
Comment (oscarbenjamin) [inline sympy/core/relational.py]: What will this do if `_rhs.is_extended_positive` is None?

Comment (JosephMehdiyev) [reply sympy/core/relational.py]: I wrote that part assuming the values are `True` or `False`, which of course is not necessarily true. Now I look at it, that parts seems very vague.
I will post a commit soon that clearly checks all 9 bool cases in that part

> What will this do if _rhs.is_extended_positive is None?

the logic (in general, not that specific part you have noted. I am not even sure if that part handles Nones properly) should return None except for the edge case ` _lhs.is_extended_positive` is True, which then should return True

**Comment by JosephMehdiyev** [comment]
I debugged one of the tests (non statistics one), seems like the issue comes from
```Python
y = Symbol('y',real=True)
Contains(Abs(y), Interval(0, oo)) # Gives True

y = Symbol('y', extended_real=True)
Contains(Abs(y), Interval(0, oo)) # Gives None, in other words the interval itself
```

Hence, 

```Python
#this works
dom = S.Reals
x = Symbol('x' )
y = Symbol('y',real=True)
solveset(Abs(x - 1) - Abs(y), x, dom) 
#This does not
y = Symbol('y')
solveset(Abs(x - 1) - Abs(y), x, dom) 
```

**Comment by JosephMehdiyev** [comment]
for the future references, the stat test that is failing is unrelated to the change, the test itself was written in a wrong way and I made an issue about this
Other failing test similar to other tests above
also the fix is not a good approach
